### PR TITLE
ISSUE-213 Implement GET /token endpoint (long live token auth only)

### PR DESCRIPTION
**Pull request by vttranlina**
REF https://github.com/linagora/tmail-backend/issues/213

### CODE REVIEW

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      .post()
+      .`then`
+      .statusCode(HttpStatus.SC_OK)
+      .extract()
+      .jsonPath()
+      .get("methodResponses[0][1].created.K38.token")
+  }
+
+  private def buildBearTokenHeader(longLivedToken: String): Header =
+    new Header(AUTHORIZATION_HEADER, "Bearer " + BOB.asString() + "_" + longLivedToken)
```
Comment (chibenwa) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala:70]: IMO this is a mis-design: we should not expect the client to know he should concatenate the two strings.

Please have `LongLivedToken/set` that can be directly passed in a Bearer header

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    val longLivedToken: String = createLongLivedToken(server)
+
+    val response: String = `given`()
+      .header(buildBearTokenHeader(longLivedToken))
+      .queryParam("type", "shortLived")
+      .queryParam("deviceId", DEVICE_ID)
+      .when()
+      .basePath(BASE_PATH)
+      .get().prettyPeek()
+      .`then`()
```
Comment (chibenwa) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala:83]: Indent for this REST call.

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/JwtTokenRoutes.scala [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+import java.util.stream
+import javax.inject.Inject
+
+case class JwtTokenRoutesModule() extends AbstractModule {
+  override def configure(): Unit = {
+    Multibinder.newSetBinder(binder, classOf[JMAPRoutes])
+      .addBinding().to(classOf[JwtTokenRoutes])
+
+    Multibinder.newSetBinder(binder, classOf[AuthenticationStrategy])
+      .addBinding().to(classOf[LongLivedTokenAuthenticationStrategy])
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/JwtTokenRoutes.scala:36]: This looks very suspicious to me. Please add a test ensuring I cannot call the main JMAP API with a long lived token.

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(response)
+      .isEqualTo(
+        """
+          |{
+          |  "token" : "$${json-unit.ignore}",
```
Comment (chibenwa) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala:94]: Have an assert ensuring I can use the main `/jmap` endpoint with that secondary token.

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/JwtTokenRoutes.scala [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+      |q8eh2eli5Qppg/Qmx1zT1W7+JYxlPsiXfmViBcY2+yNjNOHQPzyJyE+pvybW0DC+
+      |k9EJZv81OGrKInhjB/Ep6C76
+      |-----END PRIVATE KEY-----""".stripMargin
+}
+
+case class JwtTokenResponse(token: JWTToken, expiresOn: ZonedDateTime)
+
+object JwtTokenRoutes {
+  val ENDPOINT: String = "token"
+  val TOKEN_DURATION_DEFAULT: Duration = Duration.ofHours(12)
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/JwtTokenRoutes.scala:90]: No longer than 1 hour IMO.

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/JwtTokenRoutes.scala [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+  val TYPE_ACCEPT_PARAM: String = "shortLived"
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/JwtTokenRoutes.scala:91]: unused?

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+      .when()
+      .post()
+      .`then`
+      .statusCode(HttpStatus.SC_OK)
+      .extract()
+      .jsonPath()
+      .get("methodResponses[0][1].created.K38.token")
+  }
+
+  private def buildBearTokenHeader(longLivedToken: String): Header =
```
Comment (Arsnael) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala:69]: `buildBearTokenHeader`? Where is the bear? :D 

-> `buildBearerTokenHeader`

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+         |}""".stripMargin
+
+    `given`()
+      .auth()
+      .basic(BOB.asString(), BOB_PASSWORD)
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request)
+      .when()
+      .post()
+      .`then`
```
Comment (Arsnael) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala:62]: indent

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+  @BeforeEach
+  def setUp(server: GuiceJamesServer): Unit = {
+    server.getProbe(classOf[DataProbeImpl])
+      .fluent()
+      .addDomain(DOMAIN.asString())
+      .addUser(BOB.asString(), BOB_PASSWORD)
+
+    requestSpecification = baseRequestSpecBuilder(server)
+      .setAuth(new NoAuthScheme)
+      //      .setBasePath(BASE_PATH)
```
Comment (Arsnael) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala:34]: debug spotted

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+  def getTokenShouldReturnJWTTokenResponse(server: GuiceJamesServer): Unit = {
+    val longLivedToken: String = createLongLivedToken(server)
+
+    val response: String = `given`()
+      .header(buildBearTokenHeader(longLivedToken))
+      .queryParam("type", "shortLived")
+      .queryParam("deviceId", DEVICE_ID)
+      .when()
+      .basePath(BASE_PATH)
+      .get().prettyPeek()
```
Comment (Arsnael) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala:82]: debug spotted

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/longlivedtoken/LongLivedTokenInjectKeys.java [hunk 11]
```diff
@@ -0,0 +1,5 @@
+package com.linagora.tmail.james.jmap.longlivedtoken;
+
+public interface LongLivedTokenInjectKeys {
```
Comment (Arsnael) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/longlivedtoken/LongLivedTokenInjectKeys.java:3]: Java in scala repo code? Make it a trait, and you could probably just move it to `JwtTokenRoutes` as well

**Comment by Arsnael** [comment]
And rebase please :)

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/longlivedtoken/LongLivedTokenInjectKeys.java (hunk 11 above)
Comment (vttranlina) [reply tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/longlivedtoken/LongLivedTokenInjectKeys.java:3]: I have trouble with Scala, How can I create a final static String? 
When I using `@Named("JwtTokenRoutes.INJECTION_KEY")`, I got an exception: you should use a static string

Comment (chibenwa) [reply tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/longlivedtoken/LongLivedTokenInjectKeys.java:3]: Please follow this pattern : `server/protocols/jmap-rfc-8621/src/main/java/org/apache/james/jmap/http/rfc8621/InjectionKeys.java`

 -> have the java place holders in a separate Java class, java folder
 -> the rest of the code can stay in scala

Comment (vttranlina) [reply tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/longlivedtoken/LongLivedTokenInjectKeys.java]: I moved `LongLivedTokenInjectKeys` to `tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap` 
I don't sure it is right or not

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala (hunk 4 above)
Comment (vttranlina) [reply tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala:101]: I additional it at `jwtTokenResponseShouldGrantAuthToJmapEndpoint` test

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala (hunk 1 above)
Comment (vttranlina) [reply tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala]: Resolved at https://github.com/linagora/tmail-backend/pull/236

**Comment by vttranlina** [comment]
Todo
- [ ] Write code read file jwt_privateKey and initialization the Guice for `JwtPrivateKeyProvider`
- [ ] Check deviceId in the parameter is matching with longLivedToken store in the database  https://github.com/linagora/tmail-backend/issues/213#issuecomment-923773313

**Comment by vttranlina** [comment]
About todo "Write code read file jwt_privateKey and initialization the Guice for JwtPrivateKeyProvider",
Should I create a new class, that like `org.apache.james.jmap.draft.JMAPModule`  for Tmail? 
or only create provide singleton for `JwtPrivateKeyProvider` by reading configuration directly?

**Comment by chibenwa** [comment]
Do something similar to JMAPModule IMO

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala [hunk 12]
```diff
@@ -89,11 +96,142 @@ trait LinagoraLongLivedTokenAuthenticationContract {
...
+  @MethodSource(value = Array("unSupportAuthenHeaderStream"))
+  def getTokenShouldNotAcceptOtherAuthenWithoutLongLivedToken(authenHeader: Header): Unit = {
+    val response: String = `given`()
+      .header(authenHeader)
+      .queryParam("type", "shortLived")
+      .queryParam("deviceId", DEVICE_ID)
+    .when()
+      .basePath(BASE_PATH)
+    .get()
+      .`then`()
```
Comment (Arsnael) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala:116]: Wrong indent

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala [hunk 13]
```diff
@@ -0,0 +1,237 @@
...
+      .isEqualTo(
+        s"""{
+           |    "type": "about:blank",
+           |    "status": 400,
+           |    "detail": "Invalid request: 'type' must be shortLived"
+           |}""".stripMargin)
+  }
+
+  @Test
+  def getTokenShouldFailWhenMissingDeviceIdParameter(): Unit = {
```
Comment (chibenwa) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala:187]: Did you tested what happens with a bad deviceId?

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/JwtTokenRoutes.scala [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+import reactor.core.scheduler.Schedulers
+import reactor.netty.http.server.{HttpServerRequest, HttpServerResponse}
+
+import java.nio.charset.StandardCharsets
+import java.time.{Clock, Duration, ZonedDateTime}
+import java.util.stream
+import javax.inject.Inject
+import scala.jdk.CollectionConverters._
+
+case class JwtTokenRoutesModule() extends AbstractModule {
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/JwtTokenRoutes.scala:32]: ShortLivedTokenRoutesModule

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/JwtTokenRoutes.scala [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+
+object JwtTokenRoutes {
+  val ENDPOINT: String = "token"
+  val TOKEN_DURATION_DEFAULT: Duration = Duration.ofHours(1)
+  val TYPE_PARAM: String = "type"
+  val DEVICE_ID_PARAM: String = "deviceId"
+  val TYPE_ACCEPT_PARAM: String = "shortLived"
+}
+
+class JwtTokenRoutes @Inject()(@Named(LongLivedTokenInjectKeys.JMAP) val authenticator: Authenticator,
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/JwtTokenRoutes.scala:57]: ShortLivedTokenRoutes

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/JwtTokenRoutes.scala [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+      .corsHeaders())
+
+  private def generate(request: HttpServerRequest, response: HttpServerResponse): SMono[Unit] =
+    SMono(authenticator.authenticate(request))
+      .doOnNext(_ => validRequestParameter(request))
+      .map(session => {
+        val validUntil: ZonedDateTime = ZonedDateTime.ofInstant(clock.instant().plusSeconds(TOKEN_DURATION_DEFAULT.toSeconds), clock.getZone)
+        jwtSigner.sign(session.getUser, validUntil) -> validUntil
+      })
+      .map(tokenAndValidUntil => JwtTokenResponse(tokenAndValidUntil._1, tokenAndValidUntil._2))
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/JwtTokenRoutes.scala:73]:
```suggestion
      .map(session => {
        val validUntil: ZonedDateTime = ZonedDateTime.ofInstant(clock.instant().plusSeconds(TOKEN_DURATION_DEFAULT.toSeconds), clock.getZone)
         JwtTokenResponse(jwtSigner.sign(session.getUser, validUntil), validUntil)
      })
```

Get rid of the unneeded tuple?

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/JwtTokenRoutes.scala [hunk 17]
```diff
@@ -0,0 +1,125 @@
...
+      .map(JwtTokenSerializer.serializeJwtTokenResponse)
+      .map(Json.stringify)
+      .flatMap(token => SMono(response
+        .header(CONTENT_TYPE, JSON_CONTENT_TYPE)
+        .sendString(SMono.just(token))
+      ))
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/JwtTokenRoutes.scala:79]:
```suggestion
        .sendString(SMono.just(token))))
```

Picky meeeee

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala (hunk 13 above)
Comment (Arsnael) [reply tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala:187]: it's in is todo list https://github.com/linagora/tmail-backend/pull/234#issuecomment-923866977, thus why I didn't comment this :)

**Comment by vttranlina** [comment]
Regarding "Write code read file jwt_privateKey and initialization the Guice for JwtPrivateKeyProvider"
I split it into another PR (easier to follow) https://github.com/linagora/tmail-backend/pull/238

**Comment by vttranlina** [comment]
Rebase master & resolve conflict & some fixup

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/ShortLivedTokenRoutes.scala [hunk 18]
```diff
@@ -0,0 +1,154 @@
...
+  override def routes(): stream.Stream[JMAPRoute] = stream.Stream.of(
+    JMAPRoute.builder
+      .endpoint(new Endpoint(HttpMethod.GET, s"/$ENDPOINT"))
+      .action(this.generate)
+      .corsHeaders())
+
+  private def generate(request: HttpServerRequest, response: HttpServerResponse): SMono[Unit] =
+    SMono(authenticator.authenticate(request))
+      .doOnNext(_ => validRequestParameter(request))
+      .flatMap(session => {
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/ShortLivedTokenRoutes.scala:71]: We don't need `{}` here?

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/ShortLivedTokenRoutes.scala [hunk 19]
```diff
@@ -0,0 +1,154 @@
...
+      .switchIfEmpty(SMono.error(new UnauthorizedException("'deviceId' is not valid")))
+      .flatMap(secretAndDevice => validDeviceId(username, secretAndDevice._2, secretAndDevice._1))
+  }
+
+  private def extractLongLivedTokenSecretInHeader(request: HttpServerRequest): Option[LongLivedTokenSecret] =
+    Option(request.requestHeaders.get(AuthenticationStrategy.AUTHORIZATION_HEADERS))
+      .flatMap(authHeader => AuthenticationToken.fromAuthHeader(authHeader))
+      .map(authenticationToken => authenticationToken.secret)
+
+  private def validDeviceId(username: Username,
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/ShortLivedTokenRoutes.scala:104]:
```suggestion
  private def validateToken(username: Username,
```

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/ShortLivedTokenRoutes.scala [hunk 20]
```diff
@@ -0,0 +1,154 @@
...
+                            deviceId: DeviceId,
+                            secret: LongLivedTokenSecret): SMono[Unit] =
+    SMono.fromPublisher(longLivedTokenStore.validate(username, secret))
+      .map(tokenFootPrint => tokenFootPrint.deviceId.equals(deviceId))
+      .filter(isMatchingDevice => isMatchingDevice)
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/ShortLivedTokenRoutes.scala:109]:
```suggestion
      .filter(tokenFootPrint =>  tokenFootPrint.deviceId.equals(deviceId))
```

**Review by chibenwa — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
```
[INFO] deployment-tests-distributed ....................... FAILURE [06:56 min]

Caused by: java.io.FileNotFoundException: /root/conf/jwt_private (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(Unknown Source)
	at java.base/java.io.FileInputStream.<init>(Unknown Source)
	at org.apache.commons.io.FileUtils.openInputStream(FileUtils.java:2388)
	at org.apache.commons.io.FileUtils.readFileToString(FileUtils.java:2506)
	at com.linagora.tmail.james.jmap.ShortLivedTokenModule.lambda$loadKey$0(ShortLivedTokenModule.java:33)
	at com.github.fge.lambdas.functions.FunctionChainer.doApply(FunctionChainer.java:20)
	at com.github.fge.lambdas.functions.ThrowingFunction.apply(ThrowingFunction.java:17)
	... 63 more
```

Please use a volume to mount the key in there.

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — CHANGES_REQUESTED** [review]
NO, you end up putting the crypto keys in production.

Consider adding them in tests only, where needed using a docker volume in deployment tests if need be!

File: tmail-backend/apps/distributed/src/main/conf/jwt_publickey [hunk 21]
```diff
@@ -1,9 +1,9 @@
...
------END PUBLIC KEY-----
\ No newline at end of file
+MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAtlChO/nlVP27MpdkG0Bh
+16XrMRf6M4NeyGa7j5+1UKm42IKUf3lM28oe82MqIIRyvskPc11NuzSor8HmvH8H
+lhDs5DyJtx2qp35AT0zCqfwlaDnlDc/QDlZv1CoRZGpQk1Inyh6SbZwYpxxwh0fi
++d/4RpE3LBVo8wgOaXPylOlHxsDizfkL8QwXItyakBfMO6jWQRrj7/9WDhGf4Hi+
+GQur1tPGZDl9mvCoRHjFrD5M/yypIPlfMGWFVEvV5jClNMLAQ9bYFuOc7H1fEWw6
+U1LZUUbJW9/CH45YXz82CYqkrfbnQxqRb2iVbVjs/sHopHd1NTiCfUtwvcYJiBVj
+kwIDAQAB
+-----END PUBLIC KEY-----
```
Comment (chibenwa) [inline tmail-backend/apps/distributed/src/main/conf/jwt_publickey]: Can we also remove the public keys from here while we are at it?

Comment (vttranlina) [reply tmail-backend/apps/distributed/src/main/conf/jwt_publickey]: I just revert code, 
publickey already it before, I think publickey is "public", So we needn't worry about it?

Comment (chibenwa) [reply tmail-backend/apps/distributed/src/main/conf/jwt_publickey]: No we still worry about it because it grants accesses to the one with the private key: remove it too, please...

Comment (vttranlina) [reply tmail-backend/apps/distributed/src/main/conf/jwt_publickey]: I removed it.

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by vttranlina** [comment]
squash fixup commit 
& retrigger jenkin ci

**Comment by vttranlina** [comment]
Maybe we have a problem with CI, 
```
[INFO] memory-jmap-integration-tests ...................... FAILURE
```
I ran `mvn test` this module on my machine, it works

**Comment by Arsnael** [comment]
> Maybe we have a problem with CI,
> 
> ```
> [INFO] memory-jmap-integration-tests ...................... FAILURE
> ```
> 
> I ran `mvn test` this module on my machine, it works

I'm not convinced it's the CI when I see the errors. Did you rebase on the lastest master and retry those tests before saying that? :)

**Comment by vttranlina** [comment]
rebase master

**Comment by Arsnael** [comment]
I meant did you rebase and rerun tests locally? :) Because the CI does a rebase on latest master before compiling and playing the tests :)

**Comment by vttranlina** [comment]
> I meant did you rebase and rerun tests locally? 
No, I test locally with outdated code (not rebase master)

**Comment by Arsnael** [comment]
That was my point, you should test locally with rebase on latest master to see if it fails like the CI or not

**Comment by Arsnael** [comment]
If you look at the messages in the previous build:
```
[ERROR] Errors: 

[ERROR]   MemoryLongLivedTokenGetTest.getAllShouldSucceed » Provision Unable to provisio...

[ERROR]   MemoryLongLivedTokenGetTest.getShouldFailWhenMissingAllCapabilities » Provision

[ERROR]   MemoryLongLivedTokenGetTest.getShouldFailWhenMissingOneCapability » Provision ...

[ERROR]   MemoryLongLivedTokenGetTest.getShouldReturnEmptyWhenNoKey » Provision Unable t...

[ERROR]   MemoryLongLivedTokenGetTest.getShouldReturnEmptyWhenUnknownId » Provision Unab...

[ERROR]   MemoryLongLivedTokenGetTest.getShouldSucceed » Provision Unable to provision, ...
```
MemoryLongLivedTokenGetTest just got merged from @chibenwa PR. Maybe there is an update required with your modifications in your PR for this integration test to keep being valid

**Comment by Arsnael** [comment]
In `MemoryLongLivedTokenGetTest`, change L21:
```
.overrideWith(new TestJMAPServerModule()))
```
by
```
.overrideWith(new LinagoraTestJMAPServerModule()))
```
and it should be green. You might wanna check the distributed version of this integration test too

**Comment by vttranlina** [comment]
> If you look at the messages in the previous build:

yeah, I saw it on ci.
I didn't carefully read the detailed log :v

**Comment by vttranlina** [comment]
yeah, all green

**Comment by chibenwa** [comment]
Can you squash?

**Comment by vttranlina** [comment]
squash fixup
